airflow/contrib/sensors/emr_run_job_flows.py

```diff
@@ -0,0 +1,185 @@
+
+        responses = []
+        for name, job_flow_id in self.current_batch.items():
+            self.log.debug("Poking JobFlow {%s: %s}", name, job_flow_id)
+            response = emr_conn.describe_cluster(ClusterId=job_flow_id)
+            responses.append(response)
+            self._states()[name] = (job_flow_id, self._state_of(response))
+        self.log.debug("Poked JobFlow states: %s", self._states())
+
+        for failed in filter(lambda r: self._state_of(r) in
+                             EmrRunJobFlows.FAILED_STATE, responses):
+            self.log.info("there is at least one failed JobFlow")
+            return failed
+        for non_terminal in filter(lambda r: self._state_of(r) in
+                                   EmrRunJobFlows.NON_TERMINAL_STATES,
+                                   responses):
+            self.log.info("there is still at least one non-terminal JobFlow")
+            return non_terminal
+
+        # We're done with the current batch.
+        if self.job_flows:
+            self.log.info("Submitting next batch of clusters")
+            self._request_next(self.job_flows.pop(0))
+            return self.get_emr_response()
+        # All batches are in a terminal state
+        else:
+            self.log.info("Completed poking all JobFlow batches: %s",
+                          self.statuses)
+            return responses[0]
+
+    def _request_next(self, cluster_set):
+        self.current_batch = {}
+        self.statuses.append({})
+        errors = {}
+        emr_hook = EmrHook(emr_conn_id=self.emr_conn_id)
+        for name, cluster_config in cluster_set.items():
+            response = emr_hook.create_job_flow(cluster_config)
+            if response["ResponseMetadata"]["HTTPStatusCode"] != 200:
+                errors[name] = str(response)
+            else:
+                job_flow_id = response["JobFlowId"]
+                self.current_batch[name] = job_flow_id
+                self._states()[name] = (job_flow_id, "")
+        self.log.info("Requested JobFlow batch: %s", self.current_batch)
+
+        # TODO consider returning {"statuses": statuses, "errors": errors}
+        if errors:
+            self.log.error("errors: %s", errors)
+            raise AirflowException("JobFlow creation failed: " + str(errors))
+
+    def _states(self):
+        return self.statuses[-1] if self.statuses else {}
+
+    @staticmethod
+    def _state_of(response):
+        return response.get("Cluster", {}).get("Status", {}).get("State", "")
+
+    @staticmethod
+    def state_from_response(response):
+        """
+        override for EmrBaseSensor. Not using _state_of(), since
+        state_from_response expects an exception raised if the cluster State is
+        not present.
+        """
+        return EmrJobFlowSensor.state_from_response(response)
+
+    @staticmethod
+    def failure_message_from_response(response):
+        """
+        C0111
```

Comment:
This is needed for EmrBaseSensor. See `EmrJobFlowSensor.failure_message_from_response ` documentation.

```suggestion
        # pylint: disable=C0111
```